src/pages/index.js

```diff
@@ -26,92 +52,196 @@ document.addEventListener("DOMContentLoaded", () => {
     ".modal_type_profile",
-    (formData) => {
-      userInfo.setUserInfo({
-        name: formData.name,
-        description: formData.description,
-      });
-      // Close the popup after updating user info
-      profileFormPopup.close();
+    async (formData) => {
+      const name = formData.name?.trim();
+      const about = formData.description?.trim();
+
+      const originalText = profileFormPopup._submitButton.textContent;
+      profileFormPopup._submitButton.textContent = "Saving...";
+      profileFormPopup._submitButton.disabled = true;
+
+      try {
+        const updatedUser = await apiClient.patch(
+          "/users/me",
+          { name, about },
+          { headers: { authorization: AUTH_TOKEN } }
+        );
+
+        userInfo.setUserInfo({
+          name: updatedUser.name,
+          description: updatedUser.about,
+        });
+
+        profileFormPopup.close();
+      } catch (error) {
+        console.error("Failed to update profile:", error.message);
+      } finally {
+        profileFormPopup._submitButton.textContent = originalText;
+        profileFormPopup._submitButton.disabled = false;
+      }
     }
   );
   profileFormPopup.setEventListeners();
 
-  // // PopupWithForm: card creation form
-  const addCardFormPopup = new PopupWithForm(".modal_type_card", (formData) => {
-    // section.addItem(newCard.generateCard());
-    renderer({
-      name: formData.title,
+  // Popup: Add Card
+  const addCardFormPopup = new PopupWithForm(
+    ".modal_type_card",
+    async (formData) => {
+      const name = formData.title?.trim();
+      const link = formData.imageUrl?.trim();
 
-      link: formData.imageUrl,
-    });
-    // Reset the form after adding the card
-    formValidators["cardForm"].resetValidation();
-    // Close the popup after adding the card
-    addCardFormPopup.close();
-  });
+      try {
+        const createdCard = await apiClient.post(
+          "/cards",
+          { name, link },
+          { headers: { authorization: AUTH_TOKEN } }
+        );
+
+        renderer(createdCard);
+        formValidators["cardForm"].resetValidation();
+        addCardFormPopup.close();
+      } catch (error) {
+        console.error("Error creating card:", error.message);
+        throw error;
+      }
+    }
+  );
   addCardFormPopup.setEventListeners();
 
-  // Open profile form along with pre-filling data
+  // Open Profile Edit Form
   profileEditButton.addEventListener("click", () => {
     const userData = userInfo.getUserInfo();
-
-    // Pre-fill the form with current user data
     profileNameInput.value = userData.name;
     profileDescriptionInput.value = userData.description;
-
-    // Reset validation state
     formValidators["profileForm"].resetValidation();
-    // Open the profile modal
     profileFormPopup.open();
   });
 
-  // Open add card form
+  // Open Add Card Form
   addCardButton.addEventListener("click", () => {
-    // Open the add card modal
     addCardFormPopup.open();
```

Comment:
* You need to **`disable`** the submit button in the `card form` if the inputs are **empty**. It’s needed because I can click the button and add an empty card, but it should **not** happen. After the first submission, the button gets activated, and you need to **deactivate** it again. Even if I can't add an empty card, the button should look **disabled** (gray) anyway and have a `disabled` attribute.


Please, add `formValidators["cardForm"].resetValidation();` here to disable the button